src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -34,8 +37,8 @@ public class HttpFileSystemBasedFindPackageByIdResource : FindPackageByIdResourc
         private const int DefaultMaxRetries = 3;
         private int _maxRetries;
         private readonly HttpSource _httpSource;
-        private readonly ConcurrentDictionary<string, AsyncLazy<SortedDictionary<NuGetVersion, PackageInfo>>> _packageInfoCache =
-            new ConcurrentDictionary<string, AsyncLazy<SortedDictionary<NuGetVersion, PackageInfo>>>(StringComparer.OrdinalIgnoreCase);
+        private readonly ConcurrentDictionary<string, AsyncLazy<ImmutableArray<NuGetVersion>?>> _packageInfoCache =
```

Comment:
>ImmutableArray<NuGetVersion>? [](http://example.com/codeflow?start=64&length=29)

Does this need to use a nullable ImmutableArray? Could the empty array just be used in those cases?

Reply:
you're right, this can return an empty array